cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -2655,3 +2655,131 @@
        (side_effect_concat
         (x64_movrm $I64 addr_lo value_lo)
         (x64_movrm $I64 addr_hi value_hi)))))
+
+;; Rules for `load*` + ALU op + `store*` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; Add mem, reg
+(rule (lower
+       (store =flags
+              (has_type (ty_32_or_64 ty)
+                        (iadd (and
+                               (sinkable_load sink)
+                               (load flags addr offset))
```

Comment:
Strange that the first occurrence of `flags` has the `=` here rather than the second. Is that necessary?

Aside: I feel like we shouldn't even have the `=` syntax, and if you use the same variable name in multiple places we automatically unify.

Reply:
I was perplexed as well; it has to do with the inline extractor expanding with the arguments in a flipped order.

I agree that inferring `=` should be totally possible depending on whether the variable is already captured or not, and I don't know why we didn't think of that earlier! I'll follow up with a PR to allow that and see what folks think.